src/utils/jsonlogic.ts

```diff
@@ -0,0 +1,169 @@
+/**
+ * JsonLogic type checking utility functions
+ */
+import {infer} from '@open-formulieren/infernologic';
+import {type JSONObject, type JSONValue} from '@open-formulieren/types/lib/types';
+
+import {type AnyComponentSchema} from '@/types';
+
+/**
+ * @param {object | array | number | string} logic - JsonLogic expression
+ * @param {object | array | number | string} expected - example value from expected result type e.g. [["label", "value"]]
+ * @returns {string} - error message or '' if type checks
+ */
+export type JsonLogicTypeChecker = (logic: JSONValue, expected?: JSONValue) => string;
+
+type DataType =
+  | 'string'
+  | 'int'
+  | 'float'
+  | 'array'
+  | 'object'
+  | 'date'
+  | 'datetime'
+  | 'time'
+  | 'boolean';
+
+interface ServiceFetchConfiguration {
+  id: number;
+  name: string;
+  service: string;
+  path: string;
+  method: 'GET' | 'POST';
+  headers: Record<string, string>;
+  query_params: Record<string, string>;
```

Comment:
```suggestion
  query_params: Record<string, string[]>;
```
I think, because we have this array value construction: https://open-formulieren.github.io/open-forms/?path=/docs/admin-custom-mappingarrayinput--docs#mapping-to-arrays-using-valuearrayinput-argument